pkg/agent/periodicstatusupdate.go

```diff
@@ -0,0 +1,85 @@
+package agent
+
+import (
+	"time"
+
+	"github.com/Axway/agent-sdk/pkg/jobs"
+	"github.com/Axway/agent-sdk/pkg/util/errors"
+	hc "github.com/Axway/agent-sdk/pkg/util/healthcheck"
+	"github.com/Axway/agent-sdk/pkg/util/log"
+)
+
+const defaultCheckInterval time.Duration = 30000000000 // 30s
+
+type periodicStatusUpdate struct {
+	jobs.Job
+	previousActivityTime time.Time
+	currentActivityTime  time.Time
+}
+
+var statusUpdate *periodicStatusUpdate
+
+func (su *periodicStatusUpdate) Ready() bool {
+	if runStatusUpdateCheck() != nil {
+		return false
+	}
+	log.Debug("Periodic status update is ready")
+	su.currentActivityTime = time.Now()
+	su.previousActivityTime = su.currentActivityTime
+	return true
+}
+
+func (su *periodicStatusUpdate) Status() error {
+	// error out if the agent name does not exist
+	err := runStatusUpdateCheck()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (su *periodicStatusUpdate) Execute() error {
+	// error out if the agent name does not exist
+	err := runStatusUpdateCheck()
+	if err != nil {
+		log.Error(errors.ErrStatusUpdate)
+		log.Debug("Status update failed")
+		return err
+	} else {
+		// if the last timestamp for an event has changed, update the status
+		if time.Time(su.currentActivityTime).After(time.Time(su.previousActivityTime)) {
+			log.Debug("Activity change detected, updating status")
```

Comment:
can we log the times here when a change is detected since its only at debug level